src/main/java/org/devnull/jedi/PowerDNSConnectionHandler.java

```diff
@@ -125,7 +126,14 @@ public void run()
 					log.debug("waiting to read request from socket");
 				}
 
-				requestLine = reader.readLine();
+				try {
+					requestLine = reader.readLine();
+				} catch (IOException e){
```

Comment:
Actually the exception that bubbles up here is an IOException from the unix socket library.